I was running into the following error without the fix:

```
$: make dev-clean
make --no-print-directory -j8 --output-sync=line delete-hub delete-cluster1 delete-cluster2
/home/sgaddam/goprojects/downstream/multicluster-mesh-addon/bin/kind delete cluster --name hub
Deleting cluster "hub" ...
Deleted nodes: ["hub-control-plane"]
/home/sgaddam/goprojects/downstream/multicluster-mesh-addon/bin/kind delete cluster --name cluster2
Deleting cluster "cluster2" ...
failed to update kubeconfig: failed to lock config file: open /home/sgaddam/.kube/config.lock: file exists
Deleted nodes: ["cluster2-control-plane"]
ERROR: failed to delete cluster "cluster2": failed to lock config file: open /home/sgaddam/.kube/config.lock: file exists
make[1]: *** [Makefile:346: delete-cluster2] Error 1
make[1]: *** Waiting for unfinished jobs....
/home/sgaddam/goprojects/downstream/multicluster-mesh-addon/bin/kind delete cluster --name cluster1
Deleting cluster "cluster1" ...
Deleted nodes: ["cluster1-control-plane"]
make: *** [Makefile:340: dev-clean] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -337,7 +337,7 @@ dev-clean-meshes: ## Delete all mesh resources, cert-manager trust chain, and me
 
 .PHONY: dev-clean
 dev-clean: ## Destroy dev clusters and remove .kube/ folder
-	$(MAKE) --no-print-directory -j$(PARALLEL) --output-sync=line $(addprefix delete-,$(CLUSTERS))
+	$(MAKE) --no-print-directory -j1 $(addprefix delete-,$(CLUSTERS))
 	rm -rf $(DEV_KUBE_DIR)
 	$(call log,Dev environment cleaned)
 

--- a/Makefile
+++ b/Makefile
@@ -337,10 +337,10 @@ dev-clean-meshes: ## Delete all mesh resources, cert-manager trust chain, and me
 
 .PHONY: dev-clean
 dev-clean: ## Destroy dev clusters and remove .kube/ folder
-	$(MAKE) --no-print-directory -j1 $(addprefix delete-,$(CLUSTERS))
+	$(MAKE) --no-print-directory -j$(PARALLEL) --output-sync=line $(addprefix delete-,$(CLUSTERS))
 	rm -rf $(DEV_KUBE_DIR)
 	$(call log,Dev environment cleaned)
 
 .PHONY: $(addprefix delete-,$(CLUSTERS))
 $(addprefix delete-,$(CLUSTERS)): delete-%: $(KIND)
-	$(KIND) delete cluster --name $*
+	$(KIND) delete cluster --name $* --kubeconfig $(DEV_KUBE_DIR)/$*.config

